syncd/syncd_hard_reinit.cpp

```diff
@@ -323,7 +327,56 @@ sai_object_id_t processSingleVid(sai_object_id_t vid)
 
             SWSS_LOG_INFO("default virtual router will not be created, processed VID %llx to RID %llx", vid, rid);
         }
+    }
+    else if (objectType == SAI_OBJECT_TYPE_QUEUE)
+    {
+        auto it = g_vidToRidMap.find(vid);
 
+        if (it == g_vidToRidMap.end())
+        {
+            SWSS_LOG_ERROR("failed to find VID %llx in VIDTORID map", vid);
+
+            exit_and_notify(EXIT_FAILURE);
+        }
+
+        sai_object_id_t queueRid = it->second;
+
+        if (isDefaultQueueId(queueRid));
+        {
+            // this is one of default queue id
+            // we don't need to create it, just set attributes if any
+
+            rid = queueRid;
+
+            createObject = false;
+
+            SWSS_LOG_DEBUG("default queue will not be created, processed VID %llx to RID %llx", vid, rid);
+        }
+    }
+    else if (objectType == SAI_OBJECT_TYPE_PRIORITY_GROUP)
+    {
+        auto it = g_vidToRidMap.find(vid);
+
+        if (it == g_vidToRidMap.end())
+        {
+            SWSS_LOG_ERROR("failed to find VID %llx in VIDTORID map", vid);
+
+            exit_and_notify(EXIT_FAILURE);
+        }
+
+        sai_object_id_t pgRid = it->second;
+
+        if (isDefaultPriorityGroupId(pgRid))
+        {
+            // this is one of default priority group id
+            // we don't need to create it, just set attributes if any
+
+            rid = pgRid;
+
+            createObject = false;
+
+            SWSS_LOG_DEBUG("default priority group will not be created, processed VID %llx to RID %llx", vid, rid);
+        }
```

Comment:
The 3 branches above are quite similar, is it possible to reuse most of the code?